packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1840,6 +1840,7 @@ class _TabBarViewState extends State<TabBarView> {
     super.didChangeDependencies();
     _updateTabController();
     _currentIndex = _controller!.index;
+    _pageController?.dispose();
```

Comment:
actually this seems like a bug that we are recreating page controller every time. I am not sure why this is needed. My gut feeling told me this should just be `_pageController ??= PageController(...)`.

However, this seems like out of the scope of this pr. Can you add a todo here to remind us to investigate this further?